homeassistant/components/evohome/__init__.py

```diff
@@ -135,14 +141,80 @@ def _handle_exception(err) -> bool:
             )
             return False
 
-        raise  # we don't expect/handle any other ClientResponseError
+        raise  # we don't expect/handle any other Exceptions
 
 
 async def async_setup(hass: HomeAssistantType, config: ConfigType) -> bool:
     """Create a (EMEA/EU-based) Honeywell evohome system."""
-    broker = EvoBroker(hass, config[DOMAIN])
-    if not await broker.init_client():
+
+    def load_auth_tokens(app_storage) -> Tuple[Dict, Optional[Dict]]:
+        tokens = dict(app_storage if app_storage else {})
+
+        if tokens.pop(CONF_USERNAME, None) != config[DOMAIN][CONF_USERNAME]:
+            return ({}, None)  # any tokens wont be valid
+
+        # evohomeasync2 requires naive/local datetimes as strings
+        if tokens[ACCESS_TOKEN_EXPIRES] is not None:
+            tokens[ACCESS_TOKEN_EXPIRES] = _dt_to_local_naive(
+                dt_util.parse_datetime(tokens[ACCESS_TOKEN_EXPIRES])
+            )
+
+        session_data = tokens.pop(USER_DATA, None)
+
+        return (tokens, session_data)
+
+    store = hass.helpers.storage.Store(STORAGE_VERSION, STORAGE_KEY)
```

Comment:
If we start using a config entry, we can probably drop the custom storage code. Something for the future perhaps?

Reply:
Yes, it is on the radar.